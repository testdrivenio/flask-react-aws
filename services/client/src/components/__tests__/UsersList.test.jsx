import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import UsersList from "../UsersList";

afterEach(cleanup);

const users = [
  {
    email: "hermanmu@gmail.com",
    id: 1,
    username: "michael"
  },
  {
    email: "michael@mherman.org",
    id: 2,
    username: "michaelherman"
  }
];

it('renders a username', () => {
  const { getByText } = render(<UsersList users={users} removeUser={() => true} isAuthenticated={() => true} />);
  expect(getByText('michael')).toHaveClass('username');
  expect(getByText('michaelherman')).toHaveClass('username');
});

it("renders", () => {
  const { asFragment } = render(<UsersList users={users} removeUser={() => true} isAuthenticated={() => false} />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders when authenticated", () => {
  const { asFragment } = render(<UsersList users={users} removeUser={() => true} isAuthenticated={() => true} />);
  expect(asFragment()).toMatchSnapshot();
});
