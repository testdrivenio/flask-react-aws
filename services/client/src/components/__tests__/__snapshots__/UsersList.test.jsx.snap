// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <table
      class="table is-hoverable is-fullwidth"
    >
      <thead>
        <tr>
          <th>
            ID
          </th>
          <th>
            Email
          </th>
          <th>
            Username
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            hermanmu@gmail.com
          </td>
          <td
            class="username"
          >
            michael
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td>
            michael@mherman.org
          </td>
          <td
            class="username"
          >
            michaelherman
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`renders when authenticated 1`] = `
<DocumentFragment>
  <div>
    <table
      class="table is-hoverable is-fullwidth"
    >
      <thead>
        <tr>
          <th>
            ID
          </th>
          <th>
            Email
          </th>
          <th>
            Username
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            hermanmu@gmail.com
          </td>
          <td
            class="username"
          >
            michael
          </td>
          <td>
            <button
              class="button is-danger is-small"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td>
            michael@mherman.org
          </td>
          <td
            class="username"
          >
            michaelherman
          </td>
          <td>
            <button
              class="button is-danger is-small"
            >
              Delete
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
